nomos-services/cryptarchia-consensus/src/leadership.rs

```diff
@@ -1 +1,22 @@
+use cryptarchia_engine::Slot;
+use cryptarchia_ledger::{Coin, Config, EpochState, LeaderProof};
 
+pub struct Leader {
+    coins: Vec<Coin>,
+    config: cryptarchia_ledger::Config,
+}
+
+impl Leader {
+    pub fn new(coins: Vec<Coin>, config: Config) -> Self {
+        Leader { coins, config }
+    }
+
+    pub fn build_proof_for(&mut self, epoch_state: &EpochState, slot: Slot) -> Option<LeaderProof> {
+        for coin in &self.coins {
+            if coin.is_slot_leader(epoch_state, slot, &self.config.consensus_config) {
+                return Some(coin.to_proof(slot));
+            }
+        }
```

Comment:
Why do we need multiple coins in a `Leader`? I'm curious because the `Leader` in the spec has a single coin.

Reply:
Ah, I think I got to understand this while reading the cryptarchia service. A node can hold multiple coins, and if the node has at least one coin that is eligible as a leader for a specific slot, the node can propose a block. Is that right?

Reply:
Yes, it's a common scenario the one in which you have multiple coins, for example because you received multiple payments